metrics/ci_integrations/lib/source/github_actions/adapter/github_actions_source_client_adapter.dart

```diff
@@ -431,9 +449,17 @@ class GithubActionsSourceClientAdapter
     return job.completedAt.difference(job.startedAt);
   }
 
-  /// Maps the given [conclusion] to the [BuildStatus].
-  BuildStatus _mapConclusionToBuildStatus(GithubActionConclusion conclusion) {
-    switch (conclusion) {
+  /// Maps the given [job] to a [BuildStatus].
+  ///
+  /// Returns the [BuildStatus.inProgress] if the given [job]'s
+  /// [WorkflowRunJob.status] is [GithubActionStatus.inProgress]. Otherwise,
+  /// maps the [WorkflowRunJob.conclusion] to a corresponding [BuildStatus].
+  BuildStatus _mapJobToBuildStatus(WorkflowRunJob job) {
```

Comment:
Maybe it would be great to have a separate method `_getBuildStatus` or `_getJobBuildStatus` that will check whether the job has the finished status and if so, map its conclusion using the `_mapConclusionToBuildStatus` method. What do you think?